crates/chain/src/local_chain.rs

```diff
@@ -148,6 +149,34 @@ impl CheckPoint {
     pub fn iter(&self) -> CheckPointIter {
         self.clone().into_iter()
     }
+
+    /// Find checkpoint at `height`.
+    ///
+    /// Returns `None` if checkpoint at `height` does not exist`. If you are querying for multiple
+    /// heights, it will be more efficient to use [`query_multi`].
+    ///
+    /// [`query_multi`]: CheckPoint::query_multi
+    pub fn query(&self, height: u32) -> Option<Self> {
+        self.iter()
+            .take_while(|cp| cp.height() >= height)
```

Comment:
No need for this `take_while`.

Reply:
I think we need this. If we don't have this and we don't have a direct hit, we will iterate the entire chain.

Reply:
Ah good point thanks!

Reply:
Comment added.